From 88cea99fd7a41bebcaecb664827620ddaa30236c Mon Sep 17 00:00:00 2001
From: Moria Zuberi <moria.zuberi@tandemg.com>
Date: Mon, 09 Aug 2021 15:18:54 +0300
Subject: [PATCH] [SW-47873] gaudi2: update boot-info base addr

Boot-info (on SP-SRAM) address was wrong. Update to correct
address.

Change-Id: Ib7ec3a20d5350510e95097781732e5b06692606c
Signed-off-by: Moria Zuberi <moria.zuberi@tandemg.com>
(cherry picked from commit a18af5615edf381f123949d0683d90ebf8880736)
---

diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index 4c951b3..2a55053 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -25,11 +25,8 @@
 #define PPBOOT_TEXT_BASE	(FW_SRAM_BASE + 0xC)
 #define PREBOOT_TEXT_BASE	(SP_SRAM_BASE + SP_SRAM_PREBOOT_OFFSET + 0x8)
 
-/*
- * TODO: this address is wrong, a good address must be set in order to use
- * boot info API.
- */
-#define BOOT_INFO_BASE		SP_SRAM_BASE
+#define SP_SRAM_BOOT_INFO_OFFSET	0x1800
+#define BOOT_INFO_BASE		(SP_SRAM_BASE + SP_SRAM_BOOT_INFO_OFFSET)
 
 /* In order not to conflict with ARC BOOTROM's usage of DCCM,
  * Let PPBoot start from the offset of 0x402C, for using DCCM
diff --git a/arch/gaudi2/arm/common.h b/arch/gaudi2/arm/common.h
index bc5a569..aa7954d 100644
--- a/arch/gaudi2/arm/common.h
+++ b/arch/gaudi2/arm/common.h
@@ -57,7 +57,8 @@
 
 #define HL_INIT_SP_SRAM_RETRIES		50
 
-#define BOOT_INFO_BASE			HL_SP_SRAM_BASE
+#define SP_SRAM_BOOT_INFO_OFFSET	0x1800
+#define BOOT_INFO_BASE		(HL_SP_SRAM_BASE + SP_SRAM_BOOT_INFO_OFFSET)
 
 /*
  * Set ARCs in halt state before reset
